system/inc/system_exrtc.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2025 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "hal_platform.h"
+
+#if HAL_PLATFORM_EXTERNAL_RTC || HAL_PLATFORM_EXTERNAL_RTC_OPTIONAL
+
+#include "system_tick_hal.h"
+#include "am18x5.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif // __cplusplus
+
+int system_external_rtc_set_config(const particle::hal_am18x5_config_t* conf, void* reserved);
```

Comment:
I think this should be under HAL. We have for example sleep in both system and HAL, but the reason to have system-level APIs for it is that it deals with connectivity (network/cloud) and other somewhat higher level things. here you are just configuring/querying a peripheral albeit external one.